misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/NewFormattingTreeModifier.java

```diff
@@ -399,6 +429,436 @@ public RecordFieldWithDefaultValueNode transform(RecordFieldWithDefaultValueNode
+                .withWhileKeyword(whileKeyword)
+                .withCondition(condition)
+                .withWhileBody(whileBody)
+                .apply();
+    }
+
+    @Override
+    public BracedExpressionNode transform(BracedExpressionNode bracedExpressionNode) {
+        Token openParen = formatToken(bracedExpressionNode.openParen(), 0, 0);
+        ExpressionNode expression = formatNode(bracedExpressionNode.expression(), 0, 0);
+        Token closeParen = formatToken(bracedExpressionNode.closeParen(), this.trailingWS, this.trailingNL);
+
+        return bracedExpressionNode.modify()
+                .withOpenParen(openParen)
+                .withExpression(expression)
+                .withCloseParen(closeParen)
+                .apply();
+    }
+
+    @Override
+    public AssignmentStatementNode transform(AssignmentStatementNode assignmentStatementNode) {
+        Node varRef = formatNode(assignmentStatementNode.varRef(), 1, 0);
+        Token equalsToken = formatToken(assignmentStatementNode.equalsToken(), 1, 0);
+        ExpressionNode expression = formatNode(assignmentStatementNode.expression(), 0, 0);
+        Token semicolonToken = formatToken(assignmentStatementNode.semicolonToken(), this.trailingWS, this.trailingNL);
+
+        return assignmentStatementNode.modify()
+                .withVarRef(varRef)
+                .withEqualsToken(equalsToken)
+                .withExpression(expression)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public CompoundAssignmentStatementNode transform(CompoundAssignmentStatementNode compoundAssignmentStatementNode) {
+        ExpressionNode lhsExpression = formatNode(compoundAssignmentStatementNode.lhsExpression(), 1, 0);
+        Token binaryOperator = formatToken(compoundAssignmentStatementNode.binaryOperator(), 0, 0);
+        Token equalsToken = formatToken(compoundAssignmentStatementNode.equalsToken(), 1, 0);
+        ExpressionNode rhsExpression = formatNode(compoundAssignmentStatementNode.rhsExpression(), 0, 0);
+        Token semicolonToken = formatToken(compoundAssignmentStatementNode.semicolonToken(),
+                this.trailingWS, this.trailingNL);
+
+        return compoundAssignmentStatementNode.modify()
+                .withLhsExpression(lhsExpression)
+                .withBinaryOperator(binaryOperator)
+                .withEqualsToken(equalsToken)
+                .withRhsExpression(rhsExpression)
+                .withSemicolonToken(semicolonToken)
+                .apply();
+    }
+
+    @Override
+    public DoStatementNode transform(DoStatementNode doStatementNode) {
+        boolean hasOnFailClause = doStatementNode.onFailClause().isPresent();
+
+        Token doKeyword = formatToken(doStatementNode.doKeyword(), 1, 0);
+        BlockStatementNode blockStatement = formatNode(doStatementNode.blockStatement(),
+                hasOnFailClause ? 1 : this.trailingWS, hasOnFailClause ? 0 : this.trailingNL);
```

Comment:
moving this inside the below if (hasOnFailClause) { block would be cleaner..

Reply:
Done